lib/trino-record-decoder/src/test/java/io/trino/decoder/avro/TestAvroDecoder.java

```diff
@@ -378,6 +390,24 @@ public void testLongDecodedAsBigint()
         checkValue(decodedRow, row, 493857959588286460L);
     }
 
+    @Test
+    public void testIntDecodedAsDate()
+    {
+        DecoderTestColumnHandle row = new DecoderTestColumnHandle(0, "row", DATE, "id", null, null, false, false, false);
+        Map<DecoderColumnHandle, FieldValueProvider> decodedRow = buildAndDecodeColumn(row, "id", "{\"type\":\"int\",\"logicalType\":\"date\"}", 100);
+
+        checkValue(decodedRow, row, 100);
+    }
+
+    @Test
+    public void testIntDecodedAsTimeMillis()
+    {
+        DecoderTestColumnHandle row = new DecoderTestColumnHandle(0, "row", TIME_MILLIS, "id", null, null, false, false, false);
+        Map<DecoderColumnHandle, FieldValueProvider> decodedRow = buildAndDecodeColumn(row, "id", "{\"type\":\"int\",\"logicalType\":\"time-millis\"}", 100);
```

Comment:
How about for time-micros ? I guess it is also an valid Avro logical type ? Similarly for timestamp

Reply:
They can work. For the original purpose, I prepare this patch only to adapt Pulsar Trino Plugin's use case. Since the review process goes so long now, I'd prefer after a good result on the mills part and I may continue the micros part. Otherwise, it's extending the scope and I don't see an end.

BTW, Trino itself handles layout changes of TimestampType and TimeType in different PRs, so I think it's not a requirement we must do it at once.

Reply:
I do agree we can have different PRs for a dedicated datatype. But is there a way to differentiate between if the data type is of time-micros or time-mills. If a datatype is annotated with a LogicalType `time-micro` how do we restrict decoding that datatype ?

Reply:
It's the convertor's responsibility to pass the correct Trino type. `AvroColumnDecoder` accepts the Trino type the caller passes to it.